### ISSUE TITLE
Session doesn't restore fold state of all files

### ISSUE THREAD

**Issue by CheckB** [notepad-plus-plus/notepad-plus-plus#4087]
Hello, 
i have:

Windows 8.1 x64 + full update
Notepad++ 7.5.4 x64
C ++ Syntax

When I minimize something in the tray and reboot Windows, then my work tray is broken.

After rebooting, I open Notepad ++ and I see that the tray is open !!!

Again, I hide the code in the tray and immediately close notepad ++ on the X.
I open Notepad ++ and the tray is open !!!

How to solve this problem? With the opening of the tray.
Notpad I use for programming in C ++ and I constantly use the tray, but this bug repels the desire to work with notepad ++

![1](https://user-images.githubusercontent.com/34208261/34920840-76b0bc36-f98a-11e7-8ed9-a50085be1564.png)

**Comment by sasumner** [comment]
First of all, it appears that you are talking about the "code folding" feature??  Secondly, the design is that folding is not remembered across Notepad++ runs.

**Comment by CheckB** [comment]
> First of all, it appears that you are talking about the "code folding" feature??

Yes.

> Secondly, the design is that folding is not remembered across Notepad++ runs.

And how to "remember" them?

**Comment by CheckB** [comment]
There are times when you accidentally close the window.
And then you remember that there was work and there were "code folding" and you start to cry, under a tray in ~ 10k lines in 20 tabs

**Comment by sasumner** [comment]
You really should stop with the "tray" nomenclature--it is just confusing to anyone reading this that doesn't see my "code folding" comment above.  There may be a way to "script" the behavior you desire.  Let me look into that.

**Comment by dail8859** [comment]
If you want code to remain folded when you reopen Notepad++ then you need to have it save your session:
`Preferences > Backup > Remember current session for next launch`.

**Comment by CheckB** [comment]
@dail8859 

This was originally configured

![1](https://user-images.githubusercontent.com/34208261/35059357-edeac27c-fbcb-11e7-8c3b-6a055088cc83.png)

**Comment by dail8859** [comment]
Seems to work fine for just one file (which is what I was using to test with). Having more than one file seems to not restore all the file's fold state.

**Comment by CheckB** [comment]
@ dail8859 

1. Open notepad++
3. Use C++ Syntax
3. Fold some code
4. Reboot/Hard reset Windows
5. Open notepad++

**Comment by dail8859** [comment]
If you hard reset Windows don't be surprised if an application doesn't save your data.

**Comment by CheckB** [comment]
@dail8859 
So in the end, how does notepad ++ store the data?
If I used Fold some code ~ a week ago

**Comment by dail8859** [comment]
If you have a file open and close N++ it saves its state in the `session.xml` file, most likely located in your %APPDATA% directory.

It does not remember fold states of files you closed individually and open later.

**Comment by dail8859** [comment]
You are starting to get me confused...I'd suggest following the issue template so that you can adequately describe a set of reproducible steps and the issue you are running into.

**Comment by CheckB** [comment]
> The problem is that if I close notepad ++, and then open a new file (which I did not open before), then "some code" disappears - that is, not only after reboot

1. Open notepad
2. Open new file
3. Use "some code"
4. Close notepad
5. Open new file (which has never been opened)
some code is missing

**Comment by dail8859** [comment]
I am completely lost...please open a new issue and *follow the template provided* and you will have to provide more detailed information. No clue what "some code" is.
